src/routes/charities/+layout.server.ts

```diff
@@ -1,14 +1,15 @@
 import { redirect } from '@sveltejs/kit';
 import type { LayoutServerLoad } from './$types';
-import { API_BASE, getAccessTokenUrl, sampleCharities, type TCharity } from '$lib/api/charity';
+import { sampleCharities, type TCharity } from '$lib/api/charity';
+import { API_BASE } from '$lib/api/mts';
 
 export const load: LayoutServerLoad = async ({ fetch, cookies, url }) => {
 	let accessToken = url.searchParams.get('accessToken') ?? cookies.get('accessToken');
 	if (accessToken == undefined) {
 		// No access token found. Go to the MTS login page, and redirect to the same page again
 		// SvelteKit seems to default it to HTTP during dev
-		const redirectUrl = getAccessTokenUrl(url.href.replace('http:', 'https:'));
-		throw redirect(307, redirectUrl);
+		const redirectUrl = url.href.replace('http:', 'https:');
+		throw redirect(307, `/login?redirectUrl=${redirectUrl}`);
```

Comment:
Lets use `URLSearchParams` here?

Reply:
I don't think it's worth it for such a small URL, it's just one parameter